### ISSUE TITLE
Array pass by reference does not work for local array

### ISSUE THREAD

**Issue by bobbimanners** [bobbimanners/EightBall#21]
If the array is global, it works okay:

This is okay.

```
word A[10]=0
call foo(A)
end
sub foo(word A[])
  bar(A)
endsub
sub bar(word A[])
 ...
endsub
```

However, this fails:
```
call foo()
end
sub foo(word A[])
  word A[10]=0
  bar(A)
endsub
sub bar(word A[])
 ...
endsub
```
It is almost certainly a screwup with relative vs absolute addresses.

**Comment by bobbimanners** [comment]
More precisely, the problem is that taking the address of a local array in compiled code gives a *relative* address.  Needs to be an absolute address to be useful.  This works fine for scalar variables (taking address of local gives absolute address).  It also works fine in the interpreter for arrays.

**Comment by bobbimanners** [comment]
Fixed in v0.61
